### ISSUE TITLE
Rule proposal: Add fixer option for appending `|null` to type annotations of vars/params/props that have been explicitly assigned null.

### ISSUE THREAD

**Issue by MajorLift** [typescript-eslint/typescript-eslint#4663]
### Description

With strict mode on, `ts(2322)` is triggered whenever `null` is assigned to or passed in as a variable/parameter/property that has not been declared as nullable. 
e.g. `Type 'null' is not assignable to type 'ExampleType'`

An optional fixer rule for overriding this error by automatically appending `|null` to the type annotation of any such variable would be very convenient while improving type safety. 

### Fail

```ts
let replace: ExampleType = null

class ExampleClass {
  exampleProp: ExampleType
  constructor(exampleProp: ExampleType) {
    this.exampleProp = exampleProp
  }
}
const exampleClassInstance = new ExampleClass(null)
```

### Pass

```ts
let replace: ExampleType | null = null

class ExampleClass {
  exampleProp: ExampleType | null
  constructor(exampleProp: ExampleType | null) {
    this.exampleProp = exampleProp
  }
}
const exampleClassInstance = new ExampleClass(null)
```

**Comment by bradzacher** [comment]
Thanks for the request!

A few things:
1) we don't duplicate TS errors. TS does a good job of checking for type errors - we seek to augment TS's checks only.
2) there's nothing for a linter report on. TS has an error which we don't want to duplicate (as per (1)).
3) lint rules aren't for general-purpose fixers - they're for finding and reporting issues with code.
4) automated lint rule fixers must be safe to apply always. They must not break the build. Unioning in `null` is not safe as it can easily break the build by causing type errors.

This would be much better suited as an addition to the TS language server so that it suggests it as a fix when the error is encountered.
